### PR TITLE
test: update the tests to run on produced rock

### PR DESCRIPTION
**Pull request by rebornplusplus**
This PR updates the previous tests -- it runs the tests on the produced rock. The PR also updates the tests workflow to run the tests on the produced rock on push and pull_request.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -2,106 +2,39 @@ name: Tests
...
+          revision: 1552
 
       - name: Run tests
-        if: ${{ matrix.arch == 'amd64' }}
+        env:
+          ROCK_NAME: ${{ steps.build.outputs.rock }}
         run: |
           set -ex
           sudo apt-get update
           sudo apt-get install -y skopeo
```
Comment (cjdcordeiro) [inline .github/workflows/tests.yml:32]: you dont need this anymore. it's simpler and safer to use `/snap/rockcraft/current/bin/skopeo`

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]
y, that's looking good. just a minor nit

File: .github/workflows/tests.yml (hunk 1 above)
Comment (rebornplusplus) [reply .github/workflows/tests.yml]: Updated, thanks!

**Comment by rebornplusplus** [comment]
The `upload-artifact` seems to not work properly. See here: https://github.com/canonical/chiselled-python/actions/runs/7912495460?pr=16.

It's a bit confusing since we are using the exact path one step above to test the built rock. Any ideas why this might be happening?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -2,106 +2,35 @@ name: Tests
...
+        id: build
+        uses: canonical/craft-actions/rockcraft-pack@main
+        with:
+          path: python${{ matrix.version }}/
+          verbosity: debug
 
       - name: Run tests
-        if: ${{ matrix.arch == 'amd64' }}
+        env:
+          ROCK_PATH: ${{ steps.build.outputs.rock }}
```
Comment (cjdcordeiro) [inline .github/workflows/tests.yml:26]:
```suggestion
```

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -2,106 +2,35 @@ name: Tests
...
         run: |
           set -ex
-          sudo apt-get update
-          sudo apt-get install -y skopeo
-          skopeo copy oci-archive:"chiselled-python.${{ matrix.arch }}.tar" docker-daemon:${{ env.python3-8-image }}
-          ./tests/run-all-tests ${{ env.python3-8-image }}
+          ./tests/run-all-tests "$ROCK_PATH"
```
Comment (cjdcordeiro) [inline .github/workflows/tests.yml:29]:
```suggestion
          ./tests/run-all-tests "${{ steps.build.outputs.rock }}"
```

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -2,106 +2,35 @@ name: Tests
...
 
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
+        env:
+          ROCK_PATH: ${{ steps.build.outputs.rock }}
```
Comment (cjdcordeiro) [inline .github/workflows/tests.yml:33]:
```suggestion
```

File: .github/workflows/tests.yml [hunk 5]
```diff
@@ -2,106 +2,35 @@ name: Tests
...
-              "${{ secrets.MM_SERVER }}/api/v4/posts" \
-              -d @mattermost.json)"
-
-          if [[ "${HTTP_CODE}" -lt 200 || "${HTTP_CODE}" -gt 299 ]] ; then
-              echo "ERROR: unable to post message into Mattermost channel"
-              cat "$log"
-              exit 22
-          fi
+          name: rock
+          path: ./"$ROCK_PATH"
```
Comment (cjdcordeiro) [inline .github/workflows/tests.yml:36]:
```suggestion
          path: ${{ steps.build.outputs.rock }}
```

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]

**Comment by cjdcordeiro** [comment]
> The `upload-artifact` seems to not work properly. See here: https://github.com/canonical/chiselled-python/actions/runs/7912495460?pr=16.
> 
> It's a bit confusing since we are using the exact path one step above to test the built rock. Any ideas why this might be happening?

it's because the input args are not shell, so they won't accept `$` vars

**Comment by rebornplusplus** [comment]
Ah it works now, thanks!

**Review by cjdcordeiro — APPROVED** [review]
LGTM thanks

**Pull request merged**